src/mut/morphology.cpp

```diff
@@ -176,14 +175,16 @@ void Morphology::deleteSection(const std::shared_ptr<Section>& section_, bool re
             deleteSection(*it, false);
         }
     } else {
-        for (auto& child : section_->children()) {
-            // Re-link children to their "grand-parent"
-            _parent[child->id()] = _parent[id];
-            _children[_parent[id]].push_back(child);
-            if (section_->isRoot())
+        // careful not to use a ref here or you will face ref invalidation problem with vector resize
```

Comment:
```suggestion
        // Careful not to use a reference here or you will face ref invalidation problem with vector resize
        // The argument `section_` of this function could be a reference to the `_rootSections` array 
        // and that reference might be invalidated by this `push_back` (in case `vector` needs to reallocate the array)
```